frontend/src/components/PlayersBoard.js

```diff
@@ -0,0 +1,156 @@
+import styled from 'styled-components/macro';
+import ButtonTab from './ButtonTab';
+import PlayerTile from './PlayerTile';
+import ButtonBig from './ButtonBig';
+import { useState } from 'react';
+import getRandomColorCode from '../services/getRandomColorCode';
+
+export default function PlayersBoard({
+    existingPlayers,
+    setExistingPlayers,
+    selectPlayer,
+}) {
+    const [players, setPlayers] = useState([]);
+    const [playerToAdd, setPlayerToAdd] = useState('');
+    const [error, setError] = useState(false);
+    const [noPlayersSelected, setNoPlayersSelected] = useState(false);
+
+    const hasPlayerToAdd = playerToAdd.length > 0;
+
+    const handleChange = (event) => {
+        setError(false);
+        setPlayerToAdd(event.target.value);
+    };
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        if (!hasPlayerToAdd) {
+            return;
+        }
+        const playerToAddToExistingPlayers = {
+            name: playerToAdd,
+            color: getRandomColorCode(),
+        };
+        if (existingPlayers.length > 0) {
+            if (
+                !existingPlayers.find(
```

Comment:
`find` works, but `some` might be a better fit here

Reply:
alright, not sure what the difference is here, though... don't both return a boolean?